src/components/TabletApp/RightSide/QuickBook/QuickBook.tsx

```diff
@@ -4,53 +4,94 @@ import {
     Button,
     Typography,
     Grid,
-    Paper,
     TextField,
     Autocomplete,
 } from "@mui/material";
 import EditCalendarIcon from "@mui/icons-material/EditCalendar";
-import { useState, useEffect } from "react";
+import { useState } from "react";
 import axios from "axios";
-import { styled } from "@mui/material/styles";
-import { getParticipants } from "../../../../api/getRequests";
-import COLORS from "../../../../constants/CustomColors";
-
-//To be removed when globla theme is done
-const Item = styled(Paper)(({ theme }) => ({
-    backgroundColor: theme.palette.mode === "dark" ? "#1A2027" : "#fff",
-    ...theme.typography.body2,
-    padding: theme.spacing(1),
-    textAlign: "center",
-    color: theme.palette.text.secondary,
-}));
-
-//To be removed when global theme is done
-const buttonStyle = { color: COLORS.GREEN, border: `1px solid ${COLORS.GREEN}` };
+import { getParticipants, getParticipant } from "../../../../api/getRequests";
+import dayjs from "dayjs";
+import { spawnToast } from "../../../../utils/Toast";
+
+const Owner = {
+    name: "",
+    id: 0,
+    available: true,
+    image: "",
+};
+interface MeetTime {
+    end_time: string;
+    start_time: string;
+}
 
 const QuickBook = () => {
     const [timeButtonsVisible, setTimeButtonsVisible] = useState(false);
     const [open, setOpen] = useState(false);
-    const [ownerName, setOwnerName] = useState("");
+    const [owner, setOwner] = useState(Owner);
     const [owners, setOwners] = useState([""]);
     const [autoComplete, setAutoComplete] = useState(false);
+    const [time_val, setTime_val] = useState(0);
+    const [closest_meet, setMeet] = useState(0);
+    const [in_session, set_in_session] = useState(false);
 
-    const handleQuickBook = () => {
+    const handleQuickBook = async () => {
+        set_in_session(false);
         setTimeButtonsVisible(!timeButtonsVisible);
         if (timeButtonsVisible) setOpen(false);
+
+        let meetings_start_time: any[] = [];
+        const response = await axios.get("http://localhost:3001/meetings");
+
+        Object.values(response.data as MeetTime[]).map((value) => {
```

Comment:
you don't need to cast here with "as MeetTime[]", you should define what getMeetings is returning in the api/getRequests file

Reply:
defined getMeetings in api/getRequests